presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -270,8 +270,8 @@ private void processNewSources()
 
                 // check if operator is blocked
                 Operator current = operators.get(0);
-                ListenableFuture<?> blocked = isBlocked(current);
-                if (!blocked.isDone()) {
+                if (isBlocked(current)) {
+                    ListenableFuture<?> blocked = getBlockedFuture(current);
```

Comment:
I'd leave this as it was to avoid the extra creation of futures:
```java
                ListenableFuture<?> blocked = getBlockedFuture(current);
                if (!blocked.isDone()) {
```